docs/wiki/EL/devp2p.md

```diff
@@ -1,7 +1,326 @@
+- Remains unchanged across multiple sessions with the same peer.
+
+If an attacker compromises a node’s private key, past and future communications with that peer can be decrypted, making it vulnerable to long-term key exposure.
+
+###### Ephemeral-Key (Forward Secrecy)
+
+- A temporary keypair generated for each handshake, used to derive a fresh session secret.
+- Computed using ECDH between ephemeral private keys exchanged during the handshake.
+
+Since ephemeral keys are discarded after a session ends, even if an attacker later obtains a node’s long-term private key, past communications remain secure. This property is known as forward secrecy
+
+
+#### Message Framing
+
+- **Frames encapsulate encrypted messages** for efficient and secure communication.
+- **Multiplexing** allows multiple protocols to run over a single RLPx connection.
+
+##### Frame Structure
+
+| Field | Description |
+|-------|------------|
+| `header-ciphertext` | AES-encrypted **header** containing frame metadata. |
+| `header-mac` | **MAC** over the header for integrity verification. |
+| `frame-ciphertext` | AES-encrypted **message data**. |
+| `frame-mac` | **MAC** over the encrypted message data. |
+
+##### MAC Calculation
+
+- Uses **two keccak256 MAC states** (one for **ingress**, one for **egress**).
+- The MAC state is updated as frames are sent or received.
+- Ensures **message integrity** and prevents **tampering**.
+
+
+#### Capability Messaging
+
+- **Capabilities** define the supported protocols on a given connection.
+- **Multiplexing** enables concurrent usage of multiple capabilities.
+
+##### Message Structure
+
+| Field | Description |
+|-------|------------|
+| `msg-id` | Unique identifier for the message type. |
+| `msg-data` | **RLP-encoded** message payload. |
+| `frame-size` | **Compressed size** of `msg-data`. |
+
+
+#### P2P Capability Messages
+
+- The **"p2p" capability** is **mandatory** and used for initial negotiation.
+
+#### Core Messages
+
+| Message | ID | Function |
+|---------|----|----------|
+| `Hello` | `0x00` | Announces supported capabilities. |
+| `Disconnect` | `0x01` | Initiates a graceful disconnection. |
+| `Ping` | `0x02` | Checks if the peer is alive. |
+| `Pong` | `0x03` | Responds to a `Ping`. |
+
+#### Disconnect Reasons
+
+| Code | Reason |
+|------|--------|
+| `0x00` | Requested disconnect. |
+| `0x02` | Protocol violation. |
+| `0x03` | Useless peer. |
+| `0x05` | Already connected. |
+| `0x06` | Incompatible protocol version. |
+| `0x09` | Unexpected identity. |
+
+
+
+
+### 4. Application-Level Subprotocols  
+
+- **RLPx supports multiple application-level subprotocols** that enable specialized communication between Ethereum nodes.
+- These subprotocols are **built on top of the RLPx transport layer** and are used for  data exchange, state synchronization, and light client support.
+
+#### Common Ethereum Subprotocols  
```

Comment:
I think subprotocols should be in the first part of the section to provide an overview of what functions are there in the first place